fix: Avoid creating the word testfalse in the message on test timeout

DercilioFontes opened the pull request:
## Summary

Avoid creating the word "testfalse" when `takesDoneCallback` is `false` in the message printed on test timeout
Details in the issue #13844

## Test plan

Updated the timeouts test

mrazauskas commented:
Thanks! Would it be possible to a test to prevent regression in the future? Something like this one:

https://github.com/facebook/jest/blob/92630a6fadcb4a2a4b57ab4fcf6400d37594c142/e2e/__tests__/timeouts.test.ts#L125-L127

I think it should be fine to add an extra assertion to one of the tests above this one.

SimenB commented:
A test would be great, yeah 👍

DercilioFontes commented:
Ok. I’ll work on it.

DercilioFontes commented:
Hi @mrazauskas and @SimenB,

Updated the timeouts test. 

And they were false positives because of the use of `|` in the Regex to match. I fixed them and added test cases for the timeout parameter.

There is a warning about the snapshots obsolete. I don't know how to handle it.
<img width="637" alt="Screenshot 2023-02-25 at 7 24 08 PM" src="https://user-images.githubusercontent.com/20172847/221385836-3381a269-4d21-4031-aedc-fb13930fd20b.png">

DercilioFontes commented:
Checking the test failed in CircleCI.

DercilioFontes commented:
They are fixed now. I figured out the snapshots update.

SimenB reviewed (approved):
Thanks!
